lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -156,18 +166,35 @@ mReadStakeDistribution epoch db@PoolDatabase{distributions} =
     , db
     )
 
-mPutStakePoolOwner :: PoolId -> PoolOwner -> ModelPoolOp ()
-mPutStakePoolOwner poolId poolOwner db@PoolDatabase{owners} =
+mPutPoolRegistration :: PoolRegistrationCertificate -> ModelPoolOp ()
+mPutPoolRegistration registration db@PoolDatabase{owners,metadata} =
     ( Right ()
-    , db { owners = Set.insert entry owners }
+    , db { owners = Set.union entries owners
+         , metadata = Map.insert poolId (poolMargin, poolCost) metadata
+         }
     )
   where
-    entry = (poolId, poolOwner)
-
-mReadStakePoolOwners :: PoolId -> ModelPoolOp [PoolOwner]
-mReadStakePoolOwners poolId db@PoolDatabase{owners} =
-    ( Right $ L.sortOn toText
-      [ owner | (p, owner) <- Set.toList owners, p == poolId ]
+    entries = Set.map (poolId,) $ Set.fromList poolOwners
+    PoolRegistrationCertificate
+        { poolId
+        , poolOwners
+        , poolCost
+        , poolMargin
+        } = registration
+
+mReadPoolRegistration :: PoolId -> ModelPoolOp (Maybe PoolRegistrationCertificate)
+mReadPoolRegistration poolId db@PoolDatabase{owners, metadata} =
+    ( Right $ case Map.lookup poolId metadata of
+        Nothing -> Nothing
+        Just (poolMargin, poolCost) ->
+            let poolOwners = L.sortOn toText
```

Comment:
**[minor question]**

So we are sorting here on name? If yes, why not based on margin or cost?

Reply:
This is sorting the _owners_ inside **each** pool, not the pool themselves. This is really just to avoid the database (and the model) returning the list of owner in arbitrary order. This way, we can expect successive calls to return the exact same thing.